crates/router/src/connector/iatapay.rs

```diff
@@ -634,55 +632,103 @@ impl api::IncomingWebhook for Iatapay {
         &self,
         request: &api::IncomingWebhookRequestDetails<'_>,
     ) -> CustomResult<api_models::webhooks::ObjectReferenceId, errors::ConnectorError> {
-        let notif: IatapayPaymentsResponse =
-            request
-                .body
-                .parse_struct("IatapayPaymentsResponse")
-                .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
-        if notif.iata_payment_id.is_some() {
-            Ok(api_models::webhooks::ObjectReferenceId::PaymentId(
-                api_models::payments::PaymentIdType::ConnectorTransactionId(
-                    notif.iata_payment_id.unwrap_or_default(),
-                ),
-            ))
-        } else {
-            Ok(api_models::webhooks::ObjectReferenceId::RefundId(
-                api_models::webhooks::RefundIdType::ConnectorRefundId(
-                    notif.iata_refund_id.unwrap_or_default(),
-                ),
-            ))
+        let notif: iatapay::IatapayWebhookResponse = request
+            .body
+            .parse_struct("IatapayWebhookResponse")
+            .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
+        match notif {
+            iatapay::IatapayWebhookResponse::IatapayPaymentWebhookBody(wh_body) => {
+                Ok(api_models::webhooks::ObjectReferenceId::PaymentId(
```

Comment:
do we get our payment reference In response?

Reply:
yes, I have updated with merchant_payment_id (our payment id) and fall back is the iatapay_payment_id